src/Akihabara/Framework/Packet/SidePacket.cs

```diff
@@ -20,17 +20,17 @@ protected override void DeleteMpPtr()
         public int Size => SafeNativeMethods.mp_SidePacket__size(MpPtr);
 
         // According to homuler's to-do, this should be a Packet.
-        // not sure if this is correct but this should do for now.
+        // We won't bother doing that for now.
         // See: https://git.io/JcCig
-        public Packet<T> At<T>(string key)
+        public T At<T>(string key)
         {
             UnsafeNativeMethods.mp_SidePacket__at__PKc(MpPtr, key, out var packetPtr);
 
             if (packetPtr == IntPtr.Zero)
-                return default(Packet<T>);
+                return default(T);
 
             GC.KeepAlive(this);
-            return (Packet<T>)Activator.CreateInstance(typeof(Packet<T>), packetPtr, true);
+            return (T)Activator.CreateInstance(typeof(T), packetPtr, true);
```

Comment:
Is `Activator.CreateInstance()` the only way here? Reflection is a lot slower.

Reply:
If we decide to force `T` to be a `Packet` [like suggested here](https://github.com/vignetteapp/Akihabara/pull/9#issuecomment-920487957), maybe we can do something better like `new P(packetPtr, true)` or something. But in this case I don't know.

Reply:
I'll look around to see if there are any better solutions

Reply:
That would speed things up, but we need to check if `T` is always a `Packet`.

Reply:
That's suggested by Homuler's To-do but yeah I guess we'd have to verify.

Reply:
I'd say change it and then if we find any, we'll revert the commit after merge.

Reply:
What about [adding a constraint like this](https://stackoverflow.com/a/731462) to use `new`?

Reply:
Sounds good as it has reflection as a fallback.

Reply:
Alright I'll do that once I get back on my computer

Reply:
Going to veto this: We stick with the 1-to-1 behavior, any changes can be done later.

Reply:
Alright, maybe add a comment in the code linking to this thread for future reference